src/disjoint_mut.rs

```diff
@@ -54,6 +59,21 @@ pub struct DisjointMutGuard<'a, T: AsMutPtr, V: ?Sized> {
     range: (Bound<usize>, Bound<usize>),
 }
 
+impl<'a, T: AsMutPtr> DisjointMutGuard<'a, T, [u8]> {
+    pub fn cast<V: AsBytes + FromBytes>(self) -> DisjointMutGuard<'a, T, V> {
+        let mut old_guard = ManuallyDrop::new(self);
+        let bytes = mem::take(&mut old_guard.slice);
+        DisjointMutGuard {
+            slice: V::mut_from(bytes).unwrap(),
+            phantom: old_guard.phantom,
+            #[cfg(debug_assertions)]
+            parent: old_guard.parent,
+            #[cfg(debug_assertions)]
+            range: old_guard.range,
```

Comment:
Not sure if this works:

```suggestion
            phantom: old_guard.phantom,
            ..old_guard
```

Reply:
I tried this, but because the types of the two structs differ, it would require a nightly feature.

Reply:
Ahh, one day I guess we'll have nice things.